subsys/bluetooth/shell/bt.c

```diff
@@ -1970,15 +1969,18 @@ static int cmd_adv_data(const struct shell *sh, size_t argc, char *argv[])
 	atomic_set_bit_to(adv_set_opt[selected_adv], SHELL_ADV_OPT_APPEARANCE,
 			  appearance);
 
-	ad_len = ad_init(&ad[*data_len], ARRAY_SIZE(ad) - *data_len,
-			 adv_set_opt[selected_adv]);
-	if (ad_len < 0) {
+	len = ad_init(&data[*data_len], ARRAY_SIZE(ad) - *data_len, adv_set_opt[selected_adv]);
```

Comment:
If we allow this to be called with either `ad` or `sd`, then we cannot use `ARRAY_SIZE(ad)`.

Reply:
> If we allow this to be called with either `ad` or `sd`, then we cannot use `ARRAY_SIZE(ad)`.

Both `ARRAY_SIZE(ad)` or `ARRAY_SIZE(sd) ` are all same fixed value during initialization in `bt_data ad[9]` and `bt_data sd[9]`. They are just used to limit the maximum of adv elements.  

So i think it’s fun to use `ARRAY_SIZE(ad)` . Or do you think we need another parameter to replace it?

Reply:
> Both ARRAY_SIZE(ad) or ARRAY_SIZE(sd) are all same fixed value during initialization in bt_data ad[9]

Isn't that more of a coincidence than a rule? 

We could add a `#define` to set the size, and use that to initalize `ad` and `sd` and use that here as well, to ensure that they are always the same size.

Reply:
done